cmd/senpai/main.go

```diff
@@ -51,16 +53,17 @@ func main() {
 	lastNetID, lastBuffer := getLastBuffer()
 	app.SwitchToBuffer(lastNetID, lastBuffer)
 
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
gnome-terminal send SIGHUP on exit, so we should capture that as well.
```suggestion
	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
```

Reply:
NACK, see discussion above

Reply:
See my answer above https://github.com/hhirtz/senpai/pull/5#discussion_r752542452

Reply:
Well, in any case, tcell sends an error to us when the terminal closes (tty fd is closed). From my tests:

I don't know how all terminals respond to SIGTERM, but in the case of foot, kitty and gnome-terminal, they don't forward the signal (right away at least), and senpai can write the last buffer without any issue.

SIGINT shouldn't happen since the terminal mode tcell puts us in makes ^C behave like any key:
https://github.com/hhirtz/senpai/blob/06fbb04b4c0628c1befbe824be5ba333e2f3afec/app.go#L422

SIGHUP doesn't need to be handled anyway since tcell forward the error when the tty closes.

Reply:
In other words, I'm still convinced https://github.com/hhirtz/senpai/commit/06fbb04b4c0628c1befbe824be5ba333e2f3afec addresses this issue.

Reply:
There are still these cases which are worth handling:

- Clean exit on `kill senpai`
- Clean exit on system shutdown

Reply:
> tcell sends an error to us when the terminal closes (tty fd is closed)
We shouldn't rely on tcell having errors rendering when the terminal is not present, to know when the terminal was closed and we should exit. The terminal sends us a signal, let's use it.

Reply:
The tcell.Error from a closed tty is only returned on Linux. On Windows the event is never returned, we have to handle SIGHUP.